server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolAccumulator.java

```diff
@@ -470,31 +435,39 @@ public Map<String, SymbolTable> getProgramSymbols() {
    * Get variable definition node based on list of variable usage nodes.
    *
    * @param programNode the program node
-   * @param usageNodes represents variable name and its parents
+   * @param usagePath represents variable name and its parents
    * @return the list of founded variable definitions
    */
   public List<VariableNode> getVariableDefinition(
-      ProgramNode programNode, List<VariableUsageNode> usageNodes) {
-    Multimap<String, VariableNode> variables = createOrGetSymbolTable(programNode).getVariables();
+      ProgramNode programNode, List<VariableUsageNode> usagePath) {
+    SymbolTable symbolTable = createOrGetSymbolTable(programNode);
     List<VariableNode> foundDefinitions =
-        VariableUsageUtils.findVariablesForUsage(variables, usageNodes);
+            VariableUsageUtils.findVariablesForUsage(symbolTable.getVariablesMap(), usagePath);
     if (!foundDefinitions.isEmpty()) {
       return foundDefinitions;
     }
 
-    Multimap<String, VariableNode> globals = ArrayListMultimap.create();
-    getMapOfGlobalVariables(programNode)
-        .values()
-        .forEach(variableNode -> globals.put(variableNode.getName(), variableNode));
-    return VariableUsageUtils.findVariablesForUsage(globals, usageNodes);
+//    return VariableUsageUtils.findVariablesForUsage(symbolTable.getVariablesGlobalsMap(), usagePath);
+    return globalVariableSearch(symbolTable, usagePath);
   }
 
+  private List<VariableNode> globalVariableSearch(SymbolTable symbolTable, List<VariableUsageNode> usagePath) {
+    if (symbolTable.getParent() == null) {
+      return VariableUsageUtils.findVariablesForUsage(symbolTable.getVariablesGlobalsMap(), usagePath);
+    }
+    List<VariableNode> result = globalVariableSearch(symbolTable.getParent(), usagePath);
+    if (!result.isEmpty()) {
+      return result;
+    }
+    return VariableUsageUtils.findVariablesForUsage(symbolTable.getVariablesGlobalsMap(), usagePath);
+  }
   private Map<String, VariableNode> getMapOfGlobalVariables(ProgramNode programNode) {
```

Comment:
My LSP indicates that nobody actually calls this function.

Reply:
Yep, merge leftover